### PR TITLE
Improves description bfs_predecessors and bfs_successors.

### PR DESCRIPTION
**Pull request by matnagel**
Corrects the return type of bfs_predecessors.
Adds more details to the return type of bfs_successors.

Adds doctests for bfs_predecessors and bfs_successors that
are indicative for the behaviour of these functions on DiGraphs.
Previous tests only used these functions on undirect Graphs.

### CODE REVIEW

File: networkx/algorithms/traversal/breadth_first_search.py [hunk 1]
```diff
@@ -250,8 +250,8 @@ def bfs_predecessors(G, source, depth_limit=None, sort_neighbors=None):
     Returns
     -------
     pred: iterator
-        (node, predecessors) iterator where predecessors is the list of
-        predecessors of the node.
+        (node, predecessor) iterator where `predecessor` is a predecessor of
+        `node` in a shortest path from `source` to `node`.
```
Comment (dschult) [inline networkx/algorithms/traversal/breadth_first_search.py:254]:
```suggestion
        (node, predecessor) iterator where `predecessor` is the predecessor of
        `node` in a breadth first search starting from `source`.
```

File: networkx/algorithms/traversal/breadth_first_search.py [hunk 2]
```diff
@@ -310,8 +314,9 @@ def bfs_successors(G, source, depth_limit=None, sort_neighbors=None):
     Returns
     -------
     succ: iterator
-       (node, successors) iterator where successors is the list of
-       successors of the node.
+       (node, successors) iterator where `successors` is the list of
+       successors of `node`, `node` is a transitive successor of `source`
+       and `successors` is non-empty.
```
Comment (dschult) [inline networkx/algorithms/traversal/breadth_first_search.py:319]: Would it be better to define transitive successor here? Also, the first time I read this I couldn't tell if it is a list of 3 traits that successors has or if the latter phrases are simply describing other true statements. When I got to `successors is non-empty`, I decided it must be 3 ideas about successors.  I suggest splitting the sentence:

```suggestion
       (node, successors) iterator where `successors` is the non-empty list of
       successors of `node` in a breadth first search from `source`.
       To appear in the iterator,  `node` must have successors.
```

**Review by dschult — COMMENTED** [review]
Nice -- Thanks for this!

Feel to correct/change the wording in my suggestions. They aren't deal breakers.

File: networkx/algorithms/traversal/breadth_first_search.py (hunk 2 above)
Comment (matnagel) [reply networkx/algorithms/traversal/breadth_first_search.py]: Thanks! Splitting the sentence makes it more readable.

I think, we should not go into (direct) successors vs. transitive successors here. If (at all), I feel the natural place would be in the doctest of DiGraph.successors (maybe that could be another pull request).

**Comment by matnagel** [comment]
Thank you for suggestions! I have incorporated all of them.

**Review by dschult — APPROVED** [review]
Looks good to me!  Thanks!

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**